src/subsplit_dag.cpp

```diff
@@ -802,67 +832,51 @@ SizeVector SubsplitDAG::BuildEdgeReindexer(const size_t prev_edge_count) {
   // See SubsplitDAG::AddNodePair().
   for (size_t edge_idx = prev_edge_count; edge_idx < GPCSPCountWithFakeSubsplits();
        edge_idx++) {
-    size_t parent_id;
-    size_t child_id;
-    // Find the nodes that the new edge connects.
-    for (const auto &[node_pair, idx] : dag_edges_) {
-      if (idx == edge_idx) {
-        parent_id = node_pair.first;
-        child_id = node_pair.second;
-        break;
-      }
-    }
-    Bitset parent_subsplit = GetDAGNode(parent_id)->GetBitset();
-    Bitset child_subsplit = GetDAGNode(child_id)->GetBitset();
+    const auto &[node_pair, idx] =
+        *std::find_if(dag_edges_.begin(), dag_edges_.end(),
```

Comment:
This is slick 😎 

But what happens if we don't find something with edge_idx here? Should we be checking for a problem there?